registry/storage/blobwriter.go

```diff
@@ -57,6 +65,17 @@ func (bw *blobWriter) StartedAt() time.Time {
 // Commit marks the upload as completed, returning a valid descriptor. The
 // final size and digest are checked against the first descriptor provided.
 func (bw *blobWriter) Commit(ctx context.Context, desc distribution.Descriptor) (distribution.Descriptor, error) {
+	bw.mutex.Lock()
+	defer bw.closeFinishedChannel()
+	defer bw.mutex.Unlock()
```

Comment:
Should the mutex be unlocked before channel is closed or it does not matter?

Reply:
I included into the exclusion area because if a concurrent Cancel() and Commit() happens, even in that case only one of them would close the channel.

Reply:
I asked because as far as I remember, deferred functions are executed in the LIFO order - meaning that the mutex will be unlocked first and after that, `bw.closeFinishedChannel()` will be called. In other words, closing channel will be outside of the critical section.

Reply:
You are correct. Thanks for spotting this...